src/core/utils/hooks/useOverflow.test.tsx

```diff
@@ -34,21 +38,66 @@ it('should overflow when there is not enough space', () => {
     </MockComponent>,
   );
 
-  expect(container.querySelectorAll('span')).toHaveLength(4);
+  await waitFor(() => {
+    expect(container.querySelectorAll('span')).toHaveLength(4);
+  });
 
   scrollWidthSpy.mockRestore();
   offsetWidthSpy.mockRestore();
+  childOffsetWidthSpy.mockRestore();
```

Comment:
`restoreAllMocks` wont do the job? Also, move this to `afterEach`

Reply:
Changed.